android/src/main/java/com/expensify/livemarkdown/MarkdownUtils.java

```diff
@@ -81,22 +139,12 @@ public void applyMarkdownFormatting(SpannableStringBuilder ssb) {
       mPrevOutput = output;
     }
 
-    try {
-      JSONArray ranges = new JSONArray(output);
-      for (int i = 0; i < ranges.length(); i++) {
-        JSONObject range = ranges.getJSONObject(i);
-        String type = range.getString("type");
-        int start = range.getInt("start");
-        int length = range.getInt("length");
-        int depth = range.optInt("depth", 1);
-        int end = start + length;
-        if (length == 0 || end > input.length()) {
-          continue;
-        }
-        applyRange(ssb, type, start, end, depth);
+    MarkdownRange[] ranges = parseRanges(output);
+    for (MarkdownRange range: ranges) {
+      if (range.length == 0 || range.end > input.length()) {
+        continue;
       }
```

Comment:
Why do we need this? Can't we just make sure we don't add any zero-length ranges?

Reply:
Done in [`6e88d1f` (#534)](https://github.com/Expensify/react-native-live-markdown/pull/534/commits/6e88d1f33aee607ba83ab56fc3de768571cd682c)